row_cache.cc

```diff
@@ -1337,11 +1342,8 @@ std::ostream& operator<<(std::ostream& out, row_cache& rc) {
 
 future<> row_cache::do_update(row_cache::external_updater eu, row_cache::internal_updater iu) noexcept {
   // FIXME: indentation
-  return do_with(std::move(eu), std::move(iu), [this] (auto& eu, auto& iu) {
-    return futurize_invoke([this] {
-        return get_units(_update_sem, 1);
-    }).then([this, &eu, &iu] (auto permit) mutable {
-      return eu.prepare().then([this, &eu, &iu, permit = std::move(permit)] () mutable {
+    auto permit = co_await get_units(_update_sem, 1);
+      co_await eu.prepare();
```

Comment:
Can exception from prepare() abort the node because do_update() is marked as noexcept, in case we didn't defer the coroutine yet?

Reply:
Coroutines are noexcept by nature, turning all exceptions into an exceptional future return, so no terminate will happen

Reply:
This is now tested in row_cache_test.test_update_failure_in_prepare